compiler/src/dotty/tools/dotc/core/TypeApplications.scala

```diff
@@ -468,80 +421,15 @@ class TypeApplications(val self: Type) extends AnyVal {
    */
   final def safeAppliedTo(args: List[Type])(implicit ctx: Context) = self match {
     case self: TypeRef if !self.symbol.isClass && self.symbol.isCompleting =>
-      HKApply(self, args)
+      AppliedType(self, args)
     case _ =>
       appliedTo(args)
   }
 
-  /** Turn this type, which is used as an argument for
-   *  type parameter `tparam`, into a TypeBounds RHS
-   */
```

Comment:
Simplification: Everything is now handled by `baseType`, no additional methods needed.